Wrong encoding for decoding dlerror's output

### Version

3.3.0 (nightly), 3.2.3, 3.2.2, 3.2.1

### Platform

Linux x64, Linux arm64, Linux arm32

### JDK

any

### Module

LWJGL core

### Bug description

Try loading a native library that does not exist from a location that contains a non-ASCII character, or using a system locale that involves non-ASCII characters in the respective error message.

Now, look at the error message: you'll see it using characters that hint UTF-8 text being decoded using latin-1.

note: the characters of the _path_ here appear correct because the exception here is from 3.2.2, _before_ a10f862428d412087bc9a98018097aefcc9a2364. _This_ issue, however, still exists (on master):

https://github.com/LWJGL/lwjgl3/blob/ad602750ad87939276d1889b28a757061e90715b/modules/lwjgl/core/src/generated/java/org/lwjgl/system/linux/DynamicLinkLoader.java#L94-L103

It's possible this happens on macOS too, but I haven't checked for that.

### Stacktrace or crash log output

```shell
[22:55:10] [Render thread/INFO]: [STDERR]: java.lang.ExceptionInInitializerError
[22:55:10] [Render thread/INFO]: [STDERR]: 	at dvt.<init>(SourceFile:157)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at dvp.<init>(SourceFile:459)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at net.minecraft.client.main.Main.main(SourceFile:179)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:210)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:245)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at org.multimc.EntryPoint.main(EntryPoint.java:34)
[22:55:10] [Render thread/INFO]: [STDERR]: Caused by: java.lang.RuntimeException: java.lang.UnsatisfiedLinkError: Failed to dynamically load library: /home/kb1000/MultiMC5-Install/instances/pré1.18/natives/libglfw.so(error = /home/kb1000/MultiMC5-Install/instances/pré1.18/natives/libglfw.so: Kann die Shared-Object-Datei nicht Ã¶ffnen: Datei oder Verzeichnis nicht gefunden)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at dpg.<clinit>(SourceFile:46)
[22:55:10] [Render thread/INFO]: [STDERR]: 	... 11 more
[22:55:10] [Render thread/INFO]: [STDERR]: Caused by: java.lang.UnsatisfiedLinkError: Failed to dynamically load library: /home/kb1000/MultiMC5-Install/instances/pré1.18/natives/libglfw.so(error = /home/kb1000/MultiMC5-Install/instances/pré1.18/natives/libglfw.so: Kann die Shared-Object-Datei nicht Ã¶ffnen: Datei oder Verzeichnis nicht gefunden)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at org.lwjgl.system.linux.LinuxLibrary.loadLibrary(LinuxLibrary.java:32)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at org.lwjgl.system.linux.LinuxLibrary.<init>(LinuxLibrary.java:19)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at org.lwjgl.system.APIUtil.apiCreateLibrary(APIUtil.java:123)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at org.lwjgl.system.Library.loadNative(Library.java:335)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at org.lwjgl.system.Library.loadNative(Library.java:281)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at org.lwjgl.system.Library.loadNative(Library.java:205)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at org.lwjgl.glfw.GLFW.<clinit>(GLFW.java:674)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized0(Native Method)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized(Unsafe.java:1155)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at java.base/java.lang.invoke.DirectMethodHandle$EnsureInitialized.computeValue(DirectMethodHandle.java:378)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at java.base/java.lang.invoke.DirectMethodHandle$EnsureInitialized.computeValue(DirectMethodHandle.java:375)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at java.base/java.lang.ClassValue.getFromHashMap(ClassValue.java:228)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at java.base/java.lang.ClassValue.getFromBackup(ClassValue.java:210)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at java.base/java.lang.ClassValue.get(ClassValue.java:116)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at java.base/java.lang.invoke.DirectMethodHandle.checkInitialized(DirectMethodHandle.java:401)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at java.base/java.lang.invoke.DirectMethodHandle.ensureInitialized(DirectMethodHandle.java:389)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at java.base/java.lang.invoke.DirectMethodHandle.ensureInitialized(DirectMethodHandle.java:423)
[22:55:10] [Render thread/INFO]: [STDERR]: 	at dpg.<clinit>(SourceFile:43)
```

--- a/doc/notes/3.3.2.md
+++ b/doc/notes/3.3.2.md
@@ -13,6 +13,7 @@ This build includes the following changes:
 #### Fixes
 
 - Core: Fixed Java/native library incompatibility detection. (#737)
+- Core: Fixed `dlerror` decoding to use UTF-8. (#738)
 - Build: Fixed offline mode with multiple local architectures. (#740)
 
 #### Breaking Changes

--- a/modules/lwjgl/core/src/generated/java/org/lwjgl/system/linux/DynamicLinkLoader.java
+++ b/modules/lwjgl/core/src/generated/java/org/lwjgl/system/linux/DynamicLinkLoader.java
@@ -99,7 +99,7 @@ public class DynamicLinkLoader {
     @NativeType("char *")
     public static String dlerror() {
         long __result = ndlerror();
-        return memASCIISafe(__result);
+        return memUTF8Safe(__result);
     }
 
     // --- [ dlsym ] ---

--- a/modules/lwjgl/core/src/generated/java/org/lwjgl/system/macosx/DynamicLinkLoader.java
+++ b/modules/lwjgl/core/src/generated/java/org/lwjgl/system/macosx/DynamicLinkLoader.java
@@ -197,7 +197,7 @@ public class DynamicLinkLoader {
     @NativeType("char const *")
     public static String dlerror() {
         long __result = ndlerror();
-        return memASCIISafe(__result);
+        return memUTF8Safe(__result);
     }
 
     // --- [ dlsym ] ---

--- a/modules/lwjgl/core/src/templates/kotlin/core/linux/templates/DynamicLinkLoader.kt
+++ b/modules/lwjgl/core/src/templates/kotlin/core/linux/templates/DynamicLinkLoader.kt
@@ -56,7 +56,7 @@ val dlfcn = "DynamicLinkLoader".nativeClass(Module.CORE_LINUX, nativeSubPath = "
         int("mode", "a bitfield", Modes, LinkMode.BITFIELD)
     )
 
-    charASCII.p(
+    charUTF8.p(
         "dlerror",
         """
         Returns a human readable string describing the most recent error that occurred from #dlopen(), #dlsym() or #dlclose() since

--- a/modules/lwjgl/core/src/templates/kotlin/core/macos/templates/DynamicLinkLoader.kt
+++ b/modules/lwjgl/core/src/templates/kotlin/core/macos/templates/DynamicLinkLoader.kt
@@ -97,7 +97,7 @@ val dlfcn = "DynamicLinkLoader".nativeClass(Module.CORE_MACOS, nativeSubPath = "
         )
     )
 
-    charASCII.const.p(
+    charUTF8.const.p(
         "dlerror",
         """
         Provides diagnostic information corresponding to problems with calls to #dlopen(), #dlsym(), and #dlclose() in the same thread.

